ledger/ledger-api-integration-tests/src/test/lib/scala/com/digitalasset/platform/tests/integration/ledger/api/LedgerTestingHelpers.scala

```diff
@@ -56,8 +53,10 @@ class LedgerTestingHelpers(
   implicit def submitAndWait2SubmitReq(sw: SubmitAndWaitRequest): SubmitRequest =
     SubmitRequest(sw.commands, sw.traceContext)
 
-  val submitSuccessfully: SubmitRequest => Future[Assertion] =
-    submitCommand.andThen(_.map(assertCompletionIsSuccessful))
+  val submitSuccessfully: SubmitRequest => Future[Assertion] = req => {
+    Thread.sleep(45000) // FIXME this is a hack to prevent race conditions, this whole class needs to be redesigned !!
```

Comment:
this should not go in, let's discuss what the problem is

Reply:
ah, this is actually not in master. Tim put this in in his draft PR which I inherited. I have not seen the reason for this to exist yet.

Reply:
It is not needed for things to work it seems. I am not sure what was the original reason to add it.